src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Status", status)
+      ->AddVal("Power", power)
+      ->Record();
+}
+
+void Reactor::RecordInventories(double core, double reserve, double storage,
+                                double blanket, double helium) {
+  context()
+      ->NewDatum("ReactorInventories")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("TritiumCore", core)
+      ->AddVal("TritiumReserve", reserve)
+      ->AddVal("TritiumStorage", storage)
+      ->AddVal("LithiumBlanket", blanket)
+      ->AddVal("HeliumStorage", helium)
+      ->Record();
+}
+
+void Reactor::DepleteBlanket(double bred_tritium_mass) {
+  cyclus::Material::Ptr blanket_mat = blanket.Pop();
+
+  cyclus::CompMap b = blanket_mat->comp()->mass();
+  cyclus::compmath::Normalize(&b, blanket_mat->quantity());
+
+  // Percent (as decimal) of T which comes from Li-7 instead of Li-6
+  double Li7_contribution = 0.15;
+
+  cyclus::CompMap depleted_comp;
+
+  // This is ALMOST the correct behavior, but "scraping the bottom of the
+  // barrel" is a little too complex for this implementation.
+  if ((b[30060000] - (1 - Li7_contribution) * 2 * bred_tritium_mass > 0) &&
+      (b[30070000] - Li7_contribution * 7.0 / 3.0 * bred_tritium_mass > 0)) {
+    depleted_comp = {{30070000, b[30070000] - Li7_contribution * 7.0 / 3.0 *
+                                                  bred_tritium_mass},
+                     {30060000, b[30060000] - (1 - Li7_contribution) * 2 *
+                                                  bred_tritium_mass},
+                     {10030000, b[10030000] + bred_tritium_mass},
+                     {20040000, b[20040000] + 4.0 / 3.0 * bred_tritium_mass}};
+
+    // Account for the added mass of the absorbed neutrons
+    double neutron_mass_correction =
+        1.0 / 3.0 * bred_tritium_mass * (1 - Li7_contribution);
+    cyclus::Material::Ptr additional_mass = cyclus::Material::Create(
+        this, neutron_mass_correction,
+        cyclus::Composition::CreateFromMass(depleted_comp));
+
+    blanket_mat->Transmute(cyclus::Composition::CreateFromMass(depleted_comp));
+    blanket_mat->Absorb(additional_mass);
+
+    RecordOperationalInfo("Blanket Depletion",
+                          "Tritium bred at perscribed rate");
+  } else {
+    RecordOperationalInfo(
+        "Breeding Error",
+        "Blanket composition lacks sufficient lithium to continue "
+        "breeding at perscribed rate");
+  }
+  blanket.Push(blanket_mat);
+}
+
+cyclus::Material::Ptr Reactor::BreedTritium(double fuel_usage, double TBR) {
+  DepleteBlanket(fuel_usage * TBR);
+  cyclus::Material::Ptr mat = blanket.Pop();
+
+  cyclus::CompMap c = mat->comp()->mass();
+  cyclus::compmath::Normalize(&c, mat->quantity());
+
+  cyclus::CompMap T = {{10030000, 1}};
+
+  cyclus::Material::Ptr bred_fuel =
+      mat->ExtractComp(c[10030000], cyclus::Composition::CreateFromAtom(T));
```

Comment:
Make this composition once and store as a static member variable

Reply:
Done.